syntax = "proto3";

package location;

message LocationUpdate {
  string driver_id = 1;
  double lat = 2;
  double lon = 3;
  int64 timestamp = 4;
}

message Ack {
  string message = 1;
}

service LocationService {
  // Driver continuously streams location updates
  rpc StreamLocation(stream LocationUpdate) returns (Ack);
}
